Dynatrace/dynatrace-operator
# Check that generated docs and mocks are up-to-date

## Description

Our generated markdown files and mockery mocks are sometimes outdated. 
This PR adds a CI step to check if those files were updated.

In addition it: 
- removes an old check for the docs that did not work 
- update the `make doc` scripts to actually work in the CI 
- add an `issue-845-fix: True` annotation to the mockery config as the edge case anyway does not apply to our codebase
- update all outdated files

[Jira](https://dt-rnd.atlassian.net/browse/DAQ-1463)



## How can this be tested?

- Check if CI ran through
- Change one `md` file and run `act -j generated-files` locally => it should fail
